### DISCUSSION TITLE
Feature Request: Full Screen Mode

### DISCUSSION THREAD

**Discussion by ghost** [imfing/hextra#45] [Ideas]
A configuration option for full screen mode. It will use the complete width of the screen like [this theme](https://lotusdocs.dev/docs/overview/) for instance.

**Comment by imfing** [comment]
If you're asking about hiding the sidebar, I'm actually planning to add that functionality for the left sidebar soon. Stay tuned!

**Comment by ghost** [comment]
Hiding the sidebar is great!

A configuration option to use 95% - 100% of the screen width would also be good. Since more area is  used, everything (left and right sidebar, etc.) can stay in place and look [like this](https://github.com/imfing/hextra#quick-start).

**Reply by imfing** [reply]
The current page width is controlled by the  class `.max-w-screen-xl` for the outer element:
```
.max-w-screen-xl {
    max-width: 1280px;
}
```
![image](https://github.com/imfing/hextra/assets/5097752/dc8610da-ec2a-448c-bc08-db13df4f1391)

After we disable it:
<img width="2317" alt="image" src="https://github.com/imfing/hextra/assets/5097752/e5da2fc1-bb5c-4c44-b955-0bb3754901a5">

This class can be found in the `layouts`, for example:

https://github.com/imfing/hextra/blob/4553a8eda24ed3d24fa8c30cb28493e3b374e212/layouts/docs/single.html#L2

which you can override in your own layouts. 

or if you prefer changing it using custom CSS, then we can also do overriding the `max-width` of that outer `div` element.

**Reply by ghost** [reply]
Thanks Xin !

Please offer a config option whenever you have bandwidth.

**Comment by ghost** [comment]
Thanks for opening a PR !

**Reply by imfing** [reply]
I just opened an issue for future reference 😹

**Comment by imfing** [comment]
@saurabh-mish implemented in https://github.com/imfing/hextra/pull/69

**Reply by ghost** [reply]
Thanks a lot !
